community/codegen/src/main/java/org/neo4j/codegen/bytecode/ClassByteCodeWriter.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.codegen.bytecode;
+
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+
+import java.lang.reflect.Modifier;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.neo4j.codegen.ByteCodes;
+import org.neo4j.codegen.ClassEmitter;
+import org.neo4j.codegen.Expression;
+import org.neo4j.codegen.FieldReference;
+import org.neo4j.codegen.MethodDeclaration;
+import org.neo4j.codegen.MethodEmitter;
+import org.neo4j.codegen.TypeReference;
+
+import static org.neo4j.codegen.ByteCodeUtils.byteCodeName;
+import static org.neo4j.codegen.ByteCodeUtils.outerName;
+import static org.neo4j.codegen.ByteCodeUtils.signature;
+import static org.neo4j.codegen.ByteCodeUtils.typeName;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_STATIC;
+import static org.objectweb.asm.Opcodes.ACC_SUPER;
+import static org.objectweb.asm.Opcodes.PUTSTATIC;
+import static org.objectweb.asm.Opcodes.RETURN;
+import static org.objectweb.asm.Opcodes.V1_8;
+
+class ClassByteCodeWriter implements ClassEmitter
+{
+    private final ClassWriter classWriter;
+    private final TypeReference type;
+    private final Map<FieldReference,Expression> staticFields = new HashMap<>();
+    private final TypeReference base;
+
+    ClassByteCodeWriter( TypeReference type, TypeReference base, TypeReference[] interfaces )
+    {
+        this.classWriter = new ClassWriter( ClassWriter.COMPUTE_FRAMES );
```

Comment:
Does `COMPUTE_FRAMES` imply `COMPUTE_MAXS`, or should we maybe pass `COMPUTE_FRAMES | COMPUTE_MAXS` in here?